src/main/java/org/gridsuite/filter/server/TwoWindingsTransformerFilterRepositoryProxy.java

```diff
@@ -38,27 +42,42 @@ public TwoWindingsTransformerFilterRepository getRepository() {
 
     @Override
     public AbstractFilter toDto(TwoWindingsTransformerFilterEntity entity) {
-        return buildGenericFilter(
-            TwoWindingsTransformerFilter.builder()
-                .countries(AbstractFilterRepositoryProxy.cloneIfNotEmptyOrNull(entity.getCountries()))
-                .substationName(entity.getSubstationName())
-                .nominalVoltage1(AbstractFilterRepositoryProxy.convert(entity.getNominalVoltage1()))
-                .nominalVoltage2(AbstractFilterRepositoryProxy.convert(entity.getNominalVoltage2())),
-            entity).build();
+        return new FormFilter(
+            entity.getId(),
+            entity.getCreationDate(),
+            entity.getModificationDate(),
+            new TwoWindingsTransformerFilter(
+                entity.getEquipmentId(),
+                entity.getEquipmentName(),
+                entity.getSubstationName(),
+                setToSorterSet(entity.getCountries()),
+                convert(entity.getNominalVoltage1()),
+                convert(entity.getNominalVoltage2())
+                )
+        );
     }
 
     @Override
     public TwoWindingsTransformerFilterEntity fromDto(AbstractFilter dto) {
-        if (dto instanceof TwoWindingsTransformerFilter) {
-            var twoWindingsTransformerFilter = (TwoWindingsTransformerFilter) dto;
-            var twoWindingsTransformerFilterEntityBuilder = TwoWindingsTransformerFilterEntity.builder()
-                .substationName(twoWindingsTransformerFilter.getSubstationName())
-                .countries(AbstractFilterRepositoryProxy.cloneIfNotEmptyOrNull(twoWindingsTransformerFilter.getCountries()))
-                .nominalVoltage1(AbstractFilterRepositoryProxy.convert(twoWindingsTransformerFilter.getNominalVoltage1()))
-                .nominalVoltage2(AbstractFilterRepositoryProxy.convert(twoWindingsTransformerFilter.getNominalVoltage2()));
-            buildGenericFilter(twoWindingsTransformerFilterEntityBuilder, twoWindingsTransformerFilter);
-            return twoWindingsTransformerFilterEntityBuilder.build();
+        if (!(dto instanceof FormFilter)) {
+            throw new PowsyblException(WRONG_FILTER_TYPE);
         }
-        throw new PowsyblException(WRONG_FILTER_TYPE);
+        FormFilter formFilter = (FormFilter) dto;
+
+        if (!(formFilter.getEquipmentFilterForm() instanceof TwoWindingsTransformerFilter)) {
+            throw new PowsyblException(WRONG_FILTER_TYPE);
+        }
+
+        TwoWindingsTransformerFilter twoWindingsTransformerFilter = (TwoWindingsTransformerFilter) formFilter.getEquipmentFilterForm();
+        return TwoWindingsTransformerFilterEntity.builder()
+                .id(formFilter.getId())
```

Comment:
why remove the buildGenericFilter ? 
if we add another fields to genericFilter, we have to edit many files ?

Reply:
done